### PR TITLE
Create Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors …

### PR DESCRIPTION
**Pull request by bcpickett42**
…Subsystem.md

Requesting signoff for Mario Kart Bike Wireless Sensors Subsystem

### CODE REVIEW

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem 
+## Team 4 - Blake Pickett ##
+
+_Function of the Subsystem:_
+
+The function of the newly designed feature is to convert the speed and steering sensors and the microcontrollers on the Mario Kart Bike to be wireless. This new feature will mitigate safety risks to the Mario Kart Bike riders by eliminating the long wires running down the bike and around the bike stand. These long wires are a safety hazard for future customers. The wires that are rectifying and stepping down the power for the sensors on the current version of the Mario Kart Bike will be replaced with a three-AA-battery holder, complete with three AA batteries that will produce 4.5 V and 9 A current to the sensors and microcontrollers. 
+
+_Constraints:_
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: Constraints don't consider speed of communication and resolution.

Comment (JesseTNRoberts) [reply Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: Previous comment wasn't addressed. Further, I don't think you have considered how long the system will function on battery. What would be an acceptable battery life?

Comment (sagemooneyham) [reply Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: For the previous comment, I had discussed that the data throughput of the Nano 33 BLE will be around 0.27 Mbps. Is this not speed of communication? Also, I'm not sure what resolution means.

Comment (JesseTNRoberts) [reply Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: Yes, but how do you know that this is sufficient?

**Review by JesseTNRoberts — COMMENTED** [review]
How fast does this need to communicate to do what it needs to do?

**Review by JesseTNRoberts — COMMENTED** [review]
Is there not a minimum data rate necessary? 

What is the purpose of the large blocks of text in the analysis section? Analysis would include applying numerical techniques, giving simulation results, or other methods of analysis.

**Review by JesseTNRoberts — COMMENTED** [review]
Remove lengthy paragraphs.

**Review by JesseTNRoberts — COMMENTED** [review]
This looks good. I thought that D batteries were decided to be overkill. The analysis doesn't really show that they are needed.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+_Function of the Subsystem:_
+
+The function of the newly designed feature is to enable Bluetooth LE (BLE) connectivity between sensors and central Raspberry Pi (RPi). These sensors will gather, convert, and send data via BLE to the master RPi. The wires that rectify and step down the power for the sensors on the current version of the Mario Kart Bike will be replaced with four D batteries and a holder. The four D batteries will provide 6 V and 2.5 A to the microcontrollers, which will power the sensors for six hours of constant use. 
+
+_Constraints:_ 
+
+**Safety:**
+
+The long wires are a trip hazard. The long wires must be replaced with batteries.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: Why is this here?

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+_Constraints:_ 
+
+**Safety:**
+
+The long wires are a trip hazard. The long wires must be replaced with batteries. 
+
+**Run Time:**
+
+Multiple batteries must supply the DC voltage and current needed to power the equipment for 4 hours of constant use a day for 2 weeks, which is 56 hours of constant use.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: This doesn't consider the battery usage in the remaining 20*14 hours of idle time.

**Review by JesseTNRoberts — COMMENTED** [review]
The analysis is messy. It needs to be more organized.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md [hunk 4]
```diff
@@ -37,77 +34,103 @@ Communication between the sensors and main RPi must be wireless. Each sensor mus
...
+•	Max Potentiometer = 7 mA
 
-ADS1015 = 0.09091 mA 
+•	ADS1015 = 0.09091 mA 
 
-Arduino Nano 33 BLE = 15 + 2 = 17 mA
+•	Arduino Nano 33 BLE = 15 + 2 = 17 mA
 
-AA Battery Current = 2.5 A
+•	AA Battery Current = 2.5 A
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: Batteries are rated in amp hours

**Review by JesseTNRoberts — COMMENTED** [review]
Why are you not using the markdown syntax?

You can't power off the communication system.

**Comment by sagemooneyham** [comment]
> Why are you not using the markdown syntax?
> 
> You can't power off the communication system.

Why can you not power off the communication system? I will be able to power it back on in software.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+## Team 4 - Blake Pickett ##
+
+_Function of the Subsystem:_
+
+The function of the newly designed feature is to enable Bluetooth LE (BLE) connectivity between sensors and central Raspberry Pi (RPi). These sensors will gather, convert, and send data via BLE (achieved with usage of Arduino Nano 33 BLEs for each sensor) to the master RPi. The wires that rectify and step down the power for the sensors on the current version of the Mario Kart Bike will be replaced with four AA batteries and a holder. The four AA batteries will provide 6 V and 3500 mAh to the microcontrollers, which will power the sensors for more than 56 hours of constant use. 
+<br/><br/>
+
+_Constraints:_ 
+
+**Run Time:**
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: This constraint is not correct.

**Review by JesseTNRoberts — COMMENTED** [review]
The standby currents don't make sense. How are these being achieved with the current design? Why do the tach and pot not draw current when in standby? How are they put in standby?

**Review by JesseTNRoberts — COMMENTED** [review]
Previous comments stand. Account for the currents. Further, your design does not seem to support putting devices into standby mode.

**Review by JesseTNRoberts — COMMENTED** [review]
It sounds like a pair of double As would do what you need for each system. Is that accurate? Also, it seems you won't be able to wake up the microcontrollers. And you haven't analyzed the speed sensor battery usage.

File: Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+  * **_Speed Sensor Current:_**
+    
+      * Tachometer = 10 mA
+      * Arduino Nano 33 BLE Default Settings = 17 mA
+      * Arduino Nano 33 BLE Sleep Mode (Ram Retention, wake on RTC event) = 0.00316 mA  
+      * Current Leakage on IO pins (sleep) = 0.01 mA
+    <br/><br/>
+         * Current needed (running) = 10 + 17 = 27 mA
+         * Current needed (Sleeping) = 0.01 + 0.00316 = 0.01316 mA
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Mario Kart Rev 2 - Signoff - Mario Kart Bike Wireless Sensors Subsystem.md]: Where is the tachometer sleeping current coming from?

**Review by JesseTNRoberts — COMMENTED** [review]
I have the same issues as previously. The current consumed by the pot when in sleep mode is not considered and neither is the tachometer.

**Review by JesseTNRoberts — APPROVED** [review]
